src/gsy_e/models/market/two_sided.py

```diff
@@ -255,15 +255,18 @@ def accept_bid(self, bid: Bid,
         trade = Trade(str(uuid.uuid4()), self.now,
                       seller,
                       bid.buyer,
-                      bid=bid, traded_energy=energy, trade_price=trade_price,
-                      residual=residual_bid, already_tracked=already_tracked,
+                      bid=bid, offer=offer, traded_energy=energy, trade_price=trade_price,
+                      residual=residual_bid,
                       offer_bid_trade_info=updated_bid_trade_info,
                       fee_price=fee_price, time_slot=bid.time_slot,
                       matching_requirements=trade_offer_info.matching_requirements
                       )
 
-        if already_tracked is False:
-            self._update_stats_after_trade(trade, bid, already_tracked)
+        if not offer:
+            # This is a chain trade, therefore needs to be tracked. For the trade on the market
```

Comment:
What do you think about setting `offer=None` in the `accept_bid` call in the engine in order to make it explicit?

Reply:
Thanks, fixing